### PR TITLE
Refactor MTE-2009 [v123] Xcode 15.1 and iOS 17.2 upgrade

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTe-2009)

## :bulb: Description
Xcode 15.1 has been released with iOS 17.2. Let's use the latest stable version for testing.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/PhotonActionSheetTests.swift [hunk 1]
```diff
@@ -71,9 +71,9 @@ class PhotonActionSheetTests: BaseTestCase {
         mozWaitForElementToExist(app.otherElements["ActivityListView"].otherElements["example.com"])
         mozWaitForElementToExist(app.collectionViews.cells["Copy"], timeout: TIMEOUT)
 
-        var  fennecElement = app.collectionViews.scrollViews.cells.element(boundBy: 3)
+        var  fennecElement = app.collectionViews.scrollViews.cells.element(boundBy: 2)
         if iPad() {
-            fennecElement = app.collectionViews.scrollViews.cells.element(boundBy: 2)
+            fennecElement = app.collectionViews.scrollViews.cells.element(boundBy: 1)
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/PhotonActionSheetTests.swift:76]: The indexes of the icon is updated.
Simulator Screenshot - iPhone 15 Pro - 2023-12-12 at 14 48 51
Simulator Screenshot - iPad Air (5th generation) - 2023-12-12 at 15 08 19

**Comment by clarmso** [comment]
I confirmed that there's no new test failures from the full functional test suite (iPhone 15, iOS 17.2)

**Comment by isabelrios** [comment]
@clarmso there might be some test failures in the Full Functional, have you checked that?

**Comment by clarmso** [comment]
> @clarmso there might be some test failures in the Full Functional, have you checked that?

As of today, I confirmed that there's no new test failures.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/LoginsTests.swift [hunk 2]
```diff
@@ -59,7 +59,7 @@ class LoginTest: BaseTestCase {
         navigator.goto(LoginsSettings)
         unlockLoginsView()
         mozWaitForElementToExist(app.tables["Login List"])
-        XCTAssertTrue(app.searchFields["Filter"].exists)
+        XCTAssertTrue(app.searchFields["Search passwords"].exists)
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/LoginsTests.swift:62]: @isabelrios The changes in `LoginTests` seems not related to the XCode 15.1 upgrade, but I sneaked them in. 🙄

Comment (clarmso) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/LoginsTests.swift:62]: https://github.com/mozilla-mobile/firefox-ios/commit/397bea560a242f8824b11a2dbfe2187b3231f8ad

**Review by isabelrios — APPROVED** [review]
Once the conflicts are fix and build ✅ , these changes are ready to land. Thanks!
